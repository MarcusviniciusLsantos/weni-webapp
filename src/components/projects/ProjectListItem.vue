<template>
  <div class="weni-project-list-item">
    <unnnic-card-project
      :name="name"
      :action-text="$t('projects.join')"
      @action="onClick({ name: 'home', params: { projectUuid: project.uuid } })"
      :statuses="statusList"
    >
      <template slot="actions">
        <unnnic-dropdown-item
          @click="
            onClick({
              name: 'settingsProject',
              params: { projectUuid: project.uuid, internal: ['init'] },
            })
          "
        >
          <unnnic-icon-svg size="sm" icon="settings" />
          {{ $t('projects.config') }}
        </unnnic-dropdown-item>

        <unnnic-dropdown-item
          v-if="canManageMembers"
          @click="
            $store.dispatch('openRightBar', {
              props: {
                type: 'ProjectManageUsers',
                projectUuid: project.uuid,
                projectName: project.name,
                projectAuthorizations: authorizations.users,
                projectPendingAuthorizations: pendingAuthorizations.users,
                projectHasChat: hasChat,
              },

              events: {
                'added-authorization': addedAuthorization,
                'deleted-authorization': deleteUser,
                'changed-role-authorization': changedRoleAuthorization,
              },
            })
          "
        >
          <unnnic-icon-svg size="sm" icon="person" />
          {{ $t('orgs.manage_members') }}
        </unnnic-dropdown-item>

        <unnnic-dropdown-item
          v-if="canManageMembers"
          @click="
            $store.dispatch('openRightBar', {
              props: {
                type: 'ProjectSettings',
                projectUuid: project.uuid,
                projectName: project.name,
                projectDescription: project.description,
                projectTimezone: project.timezone,
                projectAuthorizations: authorizations.users,
                projectPendingAuthorizations: pendingAuthorizations.users,
                projectHasChat: hasChat,
              },

              events: {
                'added-authorization': addedAuthorization,
                'deleted-authorization': deleteUser,
                'changed-role-authorization': changedRoleAuthorization,
                'updated-project': updatedProject,
              },
            })
          "
        >
          <unnnic-icon-svg size="sm" icon="pencil-write-1" />
          {{ $t('projects.edit_name') }}
        </unnnic-dropdown-item>

        <unnnic-dropdown-item
          v-else-if="canViewMembers"
          @click="
            $store.dispatch('openRightBar', {
              props: {
                type: 'ProjectReadUsers',
                projectUuid: project.uuid,
                projectName: project.name,
                projectAuthorizations: authorizations.users,
                projectPendingAuthorizations: pendingAuthorizations.users,
                projectHasChat: hasChat,
              },

              events: {
                'deleted-authorization': deleteUser,
              },
            })
          "
        >
          <unnnic-icon-svg size="sm" icon="person" />
          {{ $t('projects.view_members') }}
        </unnnic-dropdown-item>
      </template>
    </unnnic-card-project>
  </div>
</template>

<script>
import {
  PROJECT_ROLE_MODERATOR,
  PROJECT_ROLE_CONTRIBUTOR,
} from '../users/permissionsObjects';

export default {
  name: 'ProjectListItem',

  components: {},

  props: {
    project: {
      type: Object,
    },
    name: {
      type: String,
      default: null,
    },
    time: {
      type: String,
      default: null,
    },
    aiCount: {
      type: Number,
    },
    contactCount: {
      type: Number,
    },
    flowsCount: {
      type: Number,
    },
    authorizations: {
      type: Object,
    },
    pendingAuthorizations: {
      type: Object,
    },
  },

  data() {
    return {};
  },

  created() {},

  computed: {
    canManageMembers() {
      return this.project.authorization.role === PROJECT_ROLE_MODERATOR;
    },

    canViewMembers() {
      return this.project.authorization.role === PROJECT_ROLE_CONTRIBUTOR;
    },

    hasChat() {
      return false;
    },

    statusList() {
      return [
        {
          title: this.$t('projects.ai'),
          icon: 'neurology',
          scheme: 'aux-blue',
          count: this.aiCount,
        },
        {
          title: this.$t('projects.flows'),
          icon: 'account_tree',
          scheme: 'aux-purple',
          count: this.flowsCount,
        },
        {
          title: this.$t('projects.contacts'),
          icon: 'person',
          scheme: 'aux-lemon',
          count: this.contactCount,
        },
      ];
    },
  },

  methods: {
    addedAuthorization($event) {
      this.$emit('added-authorization', $event);
    },

    onClick(route) {
      this.$emit('click', route);
    },

    async deleteUser(userEmail) {
      this.$emit('deleted-authorization', userEmail);
    },

    changedRoleAuthorization($event) {
      this.$emit('changed-role-authorization', $event);
    },
    updatedProject($event) {
      this.$emit('updated-project', $event);
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~@weni/unnnic-system/src/assets/scss/unnnic.scss';
.weni-project-list-item {
  border-radius: $unnnic-border-radius-md;
}

::v-deep .unnnic-card-project .header .buttons .unnnic-dropdown {
  user-select: none;

  .unnnic-dropdown__content {
    align-items: flex-start;

    > a {
      display: flex;
      align-items: center;
      justify-content: flex-start;
      min-width: max-content;
      line-height: $unnnic-font-size-body-md + $unnnic-line-height-md;
      width: 100%;

      & + a {
        margin-top: $unnnic-spacing-stack-sm;
      }

      > .unnnic-icon {
        margin-right: $unnnic-spacing-inline-xs;
        margin-left: 0;
      }
    }
  }
}

::v-deep .unnnic-dropdown-item + .unnnic-dropdown-item:before {
  content: none;
}
</style>
