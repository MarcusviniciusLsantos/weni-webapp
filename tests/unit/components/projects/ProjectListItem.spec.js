import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils';
import Vuex from 'vuex';
import Router from 'vue-router';
import ProjectListItem from '@/components/projects/ProjectListItem.vue';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(Router);

const router = new Router();

describe('ProjectListItem.vue', () => {
  let getters;
  let actions;
  let store;
  let wrapper;

  beforeEach(() => {
    getters = {
      currentOrg: () => {
        return {
          name: 'oi',
          uuid: '12',
          inteligence_organization: 'topp',
          authorization: 'true',
        };
      },
    };
    actions = {
      setCurrentProject: jest.fn(),
      createProject: jest.fn(),
    };

    store = new Vuex.Store({
      getters,
      actions,
    });

    wrapper = shallowMount(ProjectListItem, {
      localVue,
      i18n,
      store,
      router,
      propsData: {
        name: 'name',
        time: 'time',
        owner: 'pwmer',
        aiCount: 12,
        contactCount: 12,
        flowsCount: 12,
      },
      stubs: {
        RouterLink: RouterLinkStub,
        UnnnicIconSvg: true,
        ProjectListItem: true,
        InfiniteLoading: true,
        UnnnicDropdown: true,
        UnnnicDropdownItem: true,
        UnnnicToolTip: true,
        UnnnicTag: true,
      },
      mocks: {
        $t: () => 'some specific text',
      },
    });
  });

  it('renders a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
