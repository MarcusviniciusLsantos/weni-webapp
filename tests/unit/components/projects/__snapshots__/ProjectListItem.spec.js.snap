// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectListItem.vue renders a snapshot 1`] = `
<div class="weni-project-list-item">
  <div class="weni-project-list-item__header">
    <div class="weni-project-list-item__header__title"><span class="weni-project-list-item__header__name">
        name
      </span>
      <unnnictooltip-stub text="some specific text" side="left" enabled="true" class="weni-project-list-item__header__time__wrapper" style="display: none;"><span class="weni-project-list-item__header__time"><unnniciconsvg-stub size="xs" icon="time-clock-circle-1"></unnniciconsvg-stub>
          time
        </span></unnnictooltip-stub>
    </div>
    <div class="weni-project-list-item__header__buttons">
      <unnnictag-stub clickable="" text="some specific text" scheme="aux-blue"></unnnictag-stub>
      <unnnicdropdown-stub position="bottom-left">
        <unnniciconsvg-stub size="sm" icon="navigation-menu-vertical-1" clickable="" scheme="neutral-cleanest"></unnniciconsvg-stub>
        <unnnicdropdownitem-stub>
          <unnniciconsvg-stub size="sm" icon="cog-1"></unnniciconsvg-stub>
          some specific text
        </unnnicdropdownitem-stub>
      </unnnicdropdown-stub>
    </div>
  </div>
  <div class="weni-project-list-item__separator"></div>
  <div class="weni-project-list-item__status__list">
    <div class="weni-project-list-item__status">
      some specific text

      <div class="content">
        <div class="box aux-blue">
          <unnniciconsvg-stub size="sm" scheme="aux-blue" icon="science-fiction-robot-2"></unnniciconsvg-stub>
        </div> <span class="weni-project-list-item__status__number">
          12
        </span>
      </div>
    </div>
    <div class="weni-project-list-item__status">
      some specific text

      <div class="content">
        <div class="box aux-purple">
          <unnniciconsvg-stub size="sm" scheme="aux-purple" icon="hierarchy-3-2"></unnniciconsvg-stub>
        </div> <span class="weni-project-list-item__status__number">
          12
        </span>
      </div>
    </div>
    <div class="weni-project-list-item__status">
      some specific text

      <div class="content">
        <div class="box aux-lemon">
          <unnniciconsvg-stub size="sm" scheme="aux-lemon" icon="single-neutral-actions-1"></unnniciconsvg-stub>
        </div> <span class="weni-project-list-item__status__number">
          12
        </span>
      </div>
    </div>
  </div>
</div>
`;
